### PR TITLE
add error markers in Project Explorer

### PR DESCRIPTION
**Pull request by julsarik**
### What does this PR do?
Add error markers in the Project Explorer for resources in error as the example below :
error

### What issues does this PR fix or reference?
#7557

#### Release Notes
Add error markers in the Project Explorer.

### CODE REVIEW

**Comment by julsarik** [comment]
No tests are provided. Is this feature easily testable ?

File: plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java [hunk 1]
```diff
@@ -12,29 +12,61 @@
...
+  public PublishDiagnosticsProcessor(
+      EditorAgent editorAgent, ProjectExplorerPresenter projectExplorer) {
     this.editorAgent = editorAgent;
+    this.tree = projectExplorer.getTree();
   }
 
-  public void processDiagnostics(ExtendedPublishDiagnosticsParams diagnosticsMessage) {
-    EditorPartPresenter openedEditor =
-        editorAgent.getOpenedEditor(new Path(diagnosticsMessage.getParams().getUri()));
+  public void processDiagnostics(List<ExtendedPublishDiagnosticsParams> diagnostics) {
```
Comment (julsarik) [inline plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java]: Code server implementations should notify the UI that there is no error, sending an empty list of diagnostic.
However code server implementations currently don't notify the UI that errors have been resolved.

File: plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java [hunk 2]
```diff
@@ -12,29 +12,61 @@
...
+    tree.getNodeStorage()
+        .getAll()
+        .stream()
+        .filter(node -> node instanceof ResourceNode)
+        .forEach(
```
Comment (dkulieshov) [inline plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java]: Hi,

I guess we can add something like `.map(ResourceNode.class::cast)` here to make `forEach` section more concise.

File: plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java [hunk 3]
```diff
@@ -12,29 +12,61 @@
...
+    diagnostics.forEach(this::processDiagnostics);
+  }
+
+  private void processDiagnostics(ExtendedPublishDiagnosticsParams diagnosticsMessage) {
+    final Path path = Path.valueOf(diagnosticsMessage.getParams().getUri());
+    final Optional<Node> resource =
+        tree.getNodeStorage()
+            .getAll()
+            .stream()
+            .filter(
```
Comment (dkulieshov) [inline plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java]: Here we also can probably make it more concise and readable making some changes, something similar to this:
```
.filter(ResourceNode.class::isInstance)
.map(ResourceNode.class::cast)
.filter(node -> node.getData().isFile())
.filter(node -> path.equals(node.getData().asFile().getLocation)))
.findAny()
.ifPresent(this::setHasError);
```

What is you opinion?

**Review by dkulieshov — APPROVED** [review]

**Comment by tsmaeder** [comment]
Please consider https://github.com/eclipse/che/issues/6773 I think it's conceptually wrong to bind Diagnostics collection to open editors: There is no guarantee that diagnostics are sent when an editor is opened, they may arrive at any time.

File: plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java (hunk 3 above)
Comment (julsarik) [reply plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/PublishDiagnosticsProcessor.java]: Hi, thanks for reviewing.
Actually this is what I did in the first place but I don't know why, GWT does not compile method reference ResourceNode.class::isInstance.
I get the following error:
The type Class<ResourceNode> does not define isInstance(Node) that is applicable here

**Review by vparfonov — APPROVED** [review]

**Review by evidolob — APPROVED** [review]

**Review by vzhukovs — APPROVED** [review]

**Comment by vparfonov** [comment]
ci-test

**Comment by vparfonov** [comment]
@julsarik Thanks for your contribution, we started QA checking for you PR

**Comment by vparfonov** [comment]
ci-test

**Comment by vparfonov** [comment]
Hi, @julsarik we tried to merge your PR, but during testing found errors. TypeScript file looks ok but in project tree marked with error 
ts-error

Can you look on your PR?

**Pull request closed without merging**